notebooks/extrap-with-metadata-aggregated.ipynb

```diff
@@ -0,0 +1,257 @@
+    "\n",
+    "## 1. Import Necessary Packages\n",
+    "\n",
+    "To explore the capabilities of thicket with ExtraP, we begin by importing necessary packages."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "from IPython.display import display\n",
+    "from IPython.display import HTML\n",
+    "\n",
+    "import thicket as th\n",
+    "\n",
+    "from thicket.model_extrap import Modeling\n",
+    "\n",
+    "display(HTML(\"<style>.container { width:80% !important; }</style>\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Define Dataset Paths and Names\n",
+    "\n",
+    "In this example, we use an MPI scaling study, profiled with caliper, that has metadata about the runs. The data is also already aggregated, which means we can provide the data to Extra-P as-is."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = \"../data/mpi_scaling_cali\"\n",
+    "t_ens = th.Thicket.from_caliperreader(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Specifically, the metadata table for this set of profiles contains a `jobsize` column, which provides the amount of cores used for each profile."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "t_ens.metadata[\"jobsize\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. More Information on a Function\n",
+    "***\n",
+    "You can use the help() method within Python to see the information for a given object. You can do this by typing help(object). \n",
+    "This will allow you to see the arguments for the function, and what will be returned. An example is below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "help(Modeling)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Create Models\n",
+    "\n",
+    "First, we construct the `Modeling` object by passing all the relevant data to it. We provide `jobsize` as the `param_name` argument so the model will grab this column from the metadata table to use as our parameter. We also sub-select some metrics, since this dataset has a lot of metrics (otherwise the modeling will take a long time to do all metrics).\n",
+    "\n",
+    "Then, we call `produce_models` on that object (it's unecessary to provide an aggregation function since the data is already aggregated.).\n",
+    "\n",
+    "**NOTE:** For this example, you can view all the metric columns by adding a new cell and running: `t_ens.dataframe.columns`. "
```

Comment:
Should this be `t_ens.performance_cols`?

Reply:
I personally used t_ens.dataframe.columns to check for the columns